api/descriptions/policies/compliance_ci_images_put.md

```diff
@@ -0,0 +1,61 @@
+Updates the compliance policy for images built in your continuous integration (CI) pipeline.
+All rules in the policy are updated in a single shot.
+
+The policy set in this endpoint is enforced by the scanners in the Jenkins plugin and the `twistcli` command line tool.
+
+This endpoint maps to the policy table in **Defend > Compliance > Containers and images > CI** in the Console UI.
+
+
+### cURL Request
+
+The following cURL command overwrites all rules in your current policy with a new policy that has a single rule.
+
+Specify at least one "check", where a check is a security best practice or baseline setting to be validated by the scanner.
+Checks are specified as list in the `condition.vulnerabilities` object.
+
+For a full list of checks, go to **Defend > Compliance > Containers and images > CI** in the Console UI and create a new rule.
+All prebuilt checks and their IDs are shown under **Compliance actions**.
+
+```bash
+$ curl 'https://<CONSOLE>/api/v1/policies/compliance/ci/images' \
+  -k \
+  -X PUT \
+  -u <USER> \
+  -H 'Content-Type: application/json' \
+  -d \
+'{
+  "rules": [
+    {
+      "name": "my-rule",
+      "collections":[
+         {
+            "name":"All"
+         }
+      ],
+      "alertThreshold":{
+         "disabled":false,
+         "value":4
+      },
+      "blockThreshold":{
+         "enabled":false,
+         "value":0
+      },
+      "condition": {
+         "readonly": false,
+         "device": "",
+         "vulnerabilities": [
+         		{
+         			"id": 41,
+         			"block": false,
+         			"minSeverity": 1
+         		}
+         	]
+      }
+    }
+  ],
+  "policyType": "ciImagesCompliance"
+}'
```

Comment:
@KICKdesigns I removed the ellipses from the request body and filled in the `<VARS>`, then tested the curl command, but I get a weird result in the UI (see screenshot). Please test all curl commands. One chief customer complaints for our API docs is that the curl commands don't work, so I want to make sure we sort that out with your work.

After playing around with the command for a bit, it looks like you need specify `rules.effect`, and it's only purpose seems to be as a helper for the UI to properly fill out the **Effect** column in the policy table.

Please play around with this some more, and determine what needs to be documented (both for the curl command, endpoint description, and param description).

Here's my working command:

```
curl 'https://<CONSOLE>/api/v1/policies/compliance/ci/images' \
-k \
-X PUT \
-u <USER> \
-H 'Content-Type: application/json' \
-d \
'{
"rules": [
  {
    "name": "my-rule",
    "effect": "alert, block",
    "collections":[
       {
          "name": "All"
       }
    ],
    "condition": {
       "vulnerabilities": [
               {
                   "id": 406,
                   "block": true,
                   "minSeverity": 1
               },
               {
                   "id": 41,
                   "block": false,
                   "minSeverity": 1
               }
           ]
    }
  }
],
"policyType": "ciImagesCompliance"
}'
```

Screenshot:

<img width="1190" alt="curl-cmd-weirdness" src="https://user-images.githubusercontent.com/6518946/105514528-3098bd00-5c99-11eb-804b-c079e7c2a324.png">

Reply:
@iansk Got it thanks. When I'm testing over here, I usually go into the console UI first, then get the param curls from it (usually has a whole set that aren't required). Then I test in the command line. What is your general process?

Reply:
@KICKdesigns I usually start from the bottom up. I start with your curl command, or whatever's in the docs, and then build from there. I think you've got the right approach though. Most customers use your workflow.

Reply:
@iansk Do you want to include the 406 check in the template as well, or just the 41?

Reply:
FYI - changes have been pushed to the repo

Reply:
@KICKdesigns 

1. For all curl commands, I'd like you to describe key aspects of the command (i.e. let's do some more teaching). In this case, `effect` is a helper for the UI only. It's got no real impact on the policy. But customers should do it so that the policy table in the UI renders properly.

2. We can also say that the example command creates a rule with one check, specifically check 41.

3. Let's also explain how you to specify the various supported effects for each check (ignore, alert, fail). Explicitly including a check in `condition.vulnerabilities` and setting `condition.vulnerabilities[X].block` to `false`, sets the effect to alert. Setting `condition.vulnerabilities[X].block` to `true` sets the effect to fail. The default effect for a compliance check is ignore. That is, by **not** including it in a rule, it's assumed that the effect is ignore.

4. In your example, the value for `effect` should be `alert` only. You've only specified one check, and the check is set to alert. If you use my curl example, I've got two checks. One check is set to `alert`. One is set to `block`. So effect should list both. In the UI, these strings are a convenience so that you can quickly review the policy table and see the effect each rule has (example: you might want to quickly find the rule that's failing/blocking your build in the CI pipeline).

5. Please update bolster the relevant request params as appropriate.

Reply:
@iansk See updates. Since these generally apply to all compliance policies, do we want to move this to the general policies page? Perhaps under a subheading called "Compliance Policies"

Reply:
@KICKdesigns Yes, that's a good idea.

Reply:
@iansk See updates and new PR for the main policies page.

Reply:
@KICKdesigns LGTM